spec/typhoeus/request_spec.rb

```diff
@@ -95,6 +95,24 @@
       end
     end
 
+    context "when Config.timeout set" do
+      before { Typhoeus.configure { |config| config.timeout = 1} }
+      after { Typhoeus.configure { |config| config.timeout = 0} }
+
+      it "respects" do
+        expect(request.options[:timeout]).to be(1)
+      end
+    end
+
+    context "when Config.connecttimeout set" do
+      before { Typhoeus.configure { |config| config.connecttimeout = 1} }
+      after { Typhoeus.configure { |config| config.connecttimeout = 0} }
```

Comment:
Now it might be better if we do
```
context "when Config.connecttimeout set" do
  let!(:original_timeout_value) { Typhoeus.connecttimeout }
   before { Typhoeus.configure { |config| config.connecttimeout = 1 } }
   after { Typhoeus.configure { |config| config.connecttimeout = original_timeout_value } }
end
```

Not sure if there is a prettier way with `around` Rspec hooks.